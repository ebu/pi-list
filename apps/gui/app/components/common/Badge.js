import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from 'components/common/Icon';

const propTypes = {
    type: PropTypes.oneOf(['success', 'success2', 'danger', 'danger2', 'warning', 'warning2', 'info', 'undefined']),
    icon: PropTypes.string,
    text: PropTypes.string,
    className: PropTypes.string,
    mini: PropTypes.bool,
    border: PropTypes.bool
};

const defaultProps = {
    type: 'info',
    text: '',
    icon: '',
    className: '',
    mini: false,
    border: false
};

const Badge = (props) => {
    const badgeClassName = classNames(
        'lst-badge',
        props.className,
        {
            'lst-badge--success': props.type === 'success',
            'lst-badge--success2': props.type === 'success2',
            'lst-badge--danger': props.type === 'danger',
            'lst-badge--danger2': props.type === 'danger2',
            'lst-badge--warning': props.type === 'warning',
            'lst-badge--warning2': props.type === 'warning2',
            'lst-badge--info': props.type === 'info',
            'lst-badge--passive': props.type === 'passive',
            'lst-badge--mini': props.mini
        }
    );

    return (
        <span className={badgeClassName}>
            {props.icon && (
                <Icon value={props.icon} />
            )}
            {props.text}
        </span>
    );
};

Badge.propTypes = propTypes;
Badge.defaultProps = defaultProps;

export default Badge;
