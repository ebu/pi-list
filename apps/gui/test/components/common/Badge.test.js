import React from 'react';
import { shallow } from 'enzyme';
import Badge from 'components/common/Badge';
import renderer from 'react-test-renderer';

describe('components/common/Badge', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Badge text="Hey! It's a test"/>
        );
    });

    describe('Badge Component', () => {
        it('should include a custom className', () => {
            wrapper.setProps({ className: 'custom-classname' });

            expect(wrapper.find('.lst-badge').hasClass('custom-classname')).toBe(true);
        });

        it('should render the children test', () => {
            expect(wrapper.find('.lst-badge').text()).toBe('Hey! It\'s a test');
        });
    });

    describe('Badge Snapshots', () => {
        it('should render a default badge (info) with the correct text', () => {
            const tree = renderer
                .create(<Badge text="Just a badge" />)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a info badge with the correct text', () => {
            const tree = renderer
                .create(<Badge text="Just a badge" type="info" />)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a warning badge with the correct text', () => {
            const tree = renderer
                .create(<Badge text="Just a badge" type="warning" />)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a error badge with the correct text', () => {
            const tree = renderer
                .create(<Badge text="Just a badge" type="danger" />)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a success badge with the correct text', () => {
            const tree = renderer
                .create(<Badge text="Just a badge" type="success" />)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});
