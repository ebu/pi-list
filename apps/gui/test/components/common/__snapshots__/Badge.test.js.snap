// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Badge Badge Snapshots should render a default badge (info) with the correct text 1`] = `
<span
  className="lst-badge lst-badge--info"
>
  
  Just a badge
</span>
`;

exports[`components/common/Badge Badge Snapshots should render a error badge with the correct text 1`] = `
<span
  className="lst-badge lst-badge--danger"
>
  
  Just a badge
</span>
`;

exports[`components/common/Badge Badge Snapshots should render a info badge with the correct text 1`] = `
<span
  className="lst-badge lst-badge--info"
>
  
  Just a badge
</span>
`;

exports[`components/common/Badge Badge Snapshots should render a success badge with the correct text 1`] = `
<span
  className="lst-badge lst-badge--success"
>
  
  Just a badge
</span>
`;

exports[`components/common/Badge Badge Snapshots should render a warning badge with the correct text 1`] = `
<span
  className="lst-badge lst-badge--warning"
>
  
  Just a badge
</span>
`;
